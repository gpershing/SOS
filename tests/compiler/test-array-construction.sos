/*int array*/
//type implicit
array arr = [-2, -1, 0, 1, 2, 3]
print(arr[0]) //-2

//type explicit
array int arrr = [1]
print(arrr[0]) //1


/*float array*/
//type implicit
array arr2 = [5.2]
printf(arr2[0]) //5.2

//type explicit
array float arrr2 = [1.2, 2.4]
printf(arrr2[1]) //2.4


/*bool array*/
//type implicit
array arr3 = [True]
printb(arr3[0]) //True

//type explicit
array bool arrr3 = [False]
printb(arrr3[0]) //False


/*struct array in test-array-of-struct.sos*/


/*nested arrays*/
array array int nested = [[1], [2,3], [4,5,6]]
print(nested[0][0]) //1


/*func arrays*/
int func1() = 5
int func2(int k) = k

array arr4 = [func1(), func2(3)]
print(arr4[0]) //5
