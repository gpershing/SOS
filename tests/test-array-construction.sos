/*int array*/
arrr: array int = [1]
print(arrr[0]) //1

/*float array*/
arrr2: array float = [1.2, 2.4]
printf(arrr2[1]) //2.4


/*bool array*/
arrr3: array bool = [false]
print(arrr3[0]) //False


/*struct array in test-array-of-struct.sos*/


/*nested arrays*/
nested: array array int = [[1], [2,3], [4,5,6]]
print(nested[0][0]) //1



/*func arrays*/
add1: (k: int) -> int = k+1
minus1: (k: int) -> int = k-1
f1: func int -> int = add1
f2: func int -> int = minus1


f_usage: (i: int, f: func int -> int) -> int =
    f(i)

arr_name: array func int -> int = [f1, f2]

print(f_usage(1, arr_name[0]))
