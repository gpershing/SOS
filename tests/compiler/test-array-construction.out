-2
1
5.2
2.4
True
False
1
5
